DigitalLearningSolutions.Web/Attributes/ValidateAllowedApplicationTypeAttribute.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace DigitalLearningSolutions.Web.Attributes
+{
+    using System;
+    using System.Collections.Generic;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Filters;
+    using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    public class ValidateAllowedApplicationTypeAttribute : Attribute, IActionFilter
+    {
+        private readonly string applicationArgumentName;
+
+        public ValidateAllowedApplicationTypeAttribute(string applicationArgumentName = "application")

```

Comment:
It's up to the action which is using the attribute, I thought it makes sense for an attribute not to force something like that onto an action. Should I change it to a constant?

Reply:
Fair point. I was thinking there probably wouldn't be a case where its necessary to use a different value and it was therefore unnecessarily flexible, but it does make more sense to have to it enforced by the controller rather than the attribute. Keep it as it is.